### PR TITLE
feat(cli/tools): add a subcommand `--hide-stacktraces` for test

### PR DESCRIPTION
**Pull request by Hajime-san**
resolves https://github.com/denoland/deno/issues/24002

I implemented this suggestion that I think it sounds reasonable for reducing noise.
Also, this feature could hide stack trace when running test from editor with LSP command by internal code updated, however I don't touch it currently.

# example 

- test.ts

```ts
import { assert } from "@std/assert/assert";

Deno.test("assert a b", () => {
  assertEquals("foo", "bar");
});
```

## before

```bash
% deno test test.ts              
running 1 test from ./test.ts
assert a b ... FAILED (6ms)

 ERRORS 

assert a b => ./test.ts:3:6
error: AssertionError: Values are not equal.


    [Diff] Actual / Expected


-   foo
+   bar


  throw new AssertionError(message);
        ^
    at Module.assertEquals (https://deno.land/std@0.224.0/assert/assert_equals.ts:52:9)
    at assertEquals (https://deno.land/std@0.224.0/testing/asserts.ts:101:11)
    at file:///path/to/test.ts:4:3

 FAILURES 

assert a b => ./test.ts:3:6

FAILED | 0 passed | 1 failed (7ms)

error: Test failed
```

## after

```bash
% target/debug/deno test --hide-stacktraces test.ts               
Check file:///path/to/test.ts
running 1 test from ./test.ts
assert a b ... FAILED (10ms)

 ERRORS 

assert a b => ./test.ts:3:6
error: AssertionError: Values are not equal.


    [Diff] Actual / Expected


-   foo
+   bar



 FAILURES 

assert a b => ./test.ts:3:6

FAILED | 0 passed | 1 failed (15ms)

error: Test failed
```

### CODE REVIEW

File: cli/args/flags.rs [hunk 1]
```diff
@@ -2683,6 +2684,12 @@ Note: running multiple `deno test --clean` calls in series or parallel for the s
         .help("Select reporter to use. Default to 'pretty'.")
         .value_parser(["pretty", "dot", "junit", "tap"])
     )
+    .arg(
+      Arg::new("hide-traces")
```
Comment (lucacasonato) [inline cli/args/flags.rs:2688]: Can you rename this option to `--hide-stacktraces`?

File: cli/tools/test/mod.rs [hunk 2]
```diff
@@ -289,6 +289,11 @@ impl From<&TestDescription> for TestFailureDescription {
   }
 }
 
+#[derive(Debug, Clone, PartialEq, Deserialize)]
```
Comment (lucacasonato) [inline cli/tools/test/mod.rs:292]:
```suggestion
#[derive(Debug, Clone, PartialEq)]
```

Not needed?

**Review by lucacasonato — COMMENTED** [review]
Thanks, just a couple comments. Sorry for the slow review.

File: cli/tools/test/mod.rs (hunk 2 above)
Comment (Hajime-san) [reply cli/tools/test/mod.rs]: Oops, I forgot to delete that setting during trial and error, thanks.

**Review by dsherret — APPROVED** [review]
LGTM. Thanks!

**Comment by Hajime-san** [comment]
@dsherret 
Thank you for updating!

**Comment by dandv** [comment]
Suggestion: create a `--hide-stack-traces` alias for this flag, consistent with the multi-word hyphenation for other flags. e.g. `--permit-no-files` or `--no-clear-screen`.

**Pull request merged**